network/discovery/discovery.go

```diff
@@ -0,0 +1,427 @@
+package discovery
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/0xPolygon/polygon-edge/network/common"
+	"github.com/0xPolygon/polygon-edge/network/event"
+	"github.com/hashicorp/go-hclog"
+	"github.com/libp2p/go-libp2p-core/network"
+	"time"
+
+	"github.com/0xPolygon/polygon-edge/network/grpc"
+	"github.com/0xPolygon/polygon-edge/network/proto"
+	"github.com/libp2p/go-libp2p-core/peer"
+	kb "github.com/libp2p/go-libp2p-kbucket"
+)
+
+const (
+	// maxDiscoveryPeerReqCount is the max peer count that
+	// can be requested from other peers
+	maxDiscoveryPeerReqCount = 16
+
+	// peerDiscoveryInterval is the interval at which other
+	// peers are queried for their peer sets
+	peerDiscoveryInterval = 5 * time.Second
+
+	// bootnodeDiscoveryInterval is the interval at which
+	// random bootnodes are dialed for their peer sets
+	bootnodeDiscoveryInterval = 60 * time.Second
+)
+
+// networkingServer defines the base communication interface between
+// any networking server implementation and the DiscoveryService
+type networkingServer interface {
```

Comment:
The implementations for this interface (currently in `server.go`) should be moved in a separate .go file - just to make `server.go` less cluttered.

Reply:
Fixed 👍 

[47460e1](https://github.com/0xPolygon/polygon-edge/pull/343/commits/47460e1008876753ddd3259a9d2d52e0605c6573)